### PR TITLE
feature: show current user profile picture

### PR DESCRIPTION
**Pull request by agaldanaw2**
GitHub Issue (If applicable): #136 
## PR Type

- Feature

## What is the current behavior?

The app doesn't show the current profile picture of user

## What is the new behavior?

Show current user profile picture

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/ToDo.UI/ToDo.UI.shproj [hunk 1]
```diff
@@ -10,4 +10,7 @@
   <PropertyGroup />
   <Import Project="ToDo.UI.projitems" Label="Shared" />
   <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\CodeSharing\Microsoft.CodeSharing.CSharp.targets" />
+  <ItemGroup>
+    <_Globbed_Compile Remove="Converters\BitmapSourceConverter.cs" />
+  </ItemGroup>
```
Comment (carldebilly) [inline src/ToDo.UI/ToDo.UI.shproj:15]: This can be removed from PR

File: src/ToDo.UI/Views/HomePage.xaml [hunk 2]
```diff
@@ -12,7 +12,7 @@
 	  xmlns:utu="using:Uno.Toolkit.UI"
 	  xmlns:not_win="http://uno.ui/not_win"
 	  Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
-	  mc:Ignorable="d not_win">
```
Comment (carldebilly) [inline src/ToDo.UI/Views/HomePage.xaml:15]: It is a common practice to keep this `d:` namespace.

File: src/ToDo/Presentation/WelcomeViewModel.cs [hunk 3]
```diff
@@ -4,27 +4,33 @@
...
 	{
 		var user = await _authService.AuthenticateAsync(_dispatcher);
+		if (user is null) return;
 
-		if(user is not null)
+		var profile = await _userSvc.GetAsync(ct);
+		if (profile is not null)
 		{
-			await _navigator.NavigateRouteAsync(this, string.Empty, cancellation: ct);
+			_authService.SetProfilePicture(profile);
```
Comment (carldebilly) [inline src/ToDo/Presentation/WelcomeViewModel.cs:32]: This is a strange pattern, and it shouldn't be the responsibility of a VM to set the state of a business service, especially when the source is an endpoint.

File: src/ToDo/ToDo.csproj [hunk 4]
```diff
@@ -4,6 +4,9 @@
 		<ImplicitUsings>enable</ImplicitUsings>
 		<Nullable>enable</Nullable>
 	</PropertyGroup>
+	<ItemGroup>
+	  <None Remove="Presentation\HomeViewModel.cs~RFf0f4d75.TMP" />
+	</ItemGroup>
```
Comment (carldebilly) [inline src/ToDo/ToDo.csproj:9]: This should not be in the PR.

File: src/ToDo/Business/Mock/MockAuthenticationService.cs [hunk 5]
```diff
@@ -24,4 +24,13 @@ public async Task SignOutAsync()
 	{
 		_user = null;
 	}
+
+	public async Task SetProfilePicture(IUserProfilePictureService userProfilePictureService, CancellationToken cancellation)
```
Comment (Xiaoy312) [inline src/ToDo/Business/Mock/MockAuthenticationService.cs:28]: do the same as AuthenticationService

File: src/ToDo/Business/Models/UserContext.cs [hunk 6]
```diff
@@ -7,4 +7,5 @@ public record UserContext
 	public string? Email { get; init; }
 
 	public string? AccessToken { get; init; }
+	public byte[]? ProfilePicture { get; init; }
```
Comment (Xiaoy312) [inline src/ToDo/Business/Models/UserContext.cs:10]: minor nitpicking: you see the existing code have empty line between members, try to follow the same style as well
gotta keep the coding style consistent

File: src/ToDo.UI/Views/HomePage.xaml [hunk 7]
```diff
@@ -98,37 +98,37 @@
...
+			<!-- Task Lists: template for WellKnownLists & CustomLists -->
+			<muxc:NavigationView.MenuItemTemplate>
+				<DataTemplate>
+					<muxc:NavigationViewItem Content="{Binding DisplayName}"
 										 uen:Navigation.Data="{Binding}">
-					<muxc:NavigationViewItem.Icon>
-						<PathIcon Data="{Binding Converter={StaticResource TaskListIconValueConverter}}"
+						<muxc:NavigationViewItem.Icon>
+							<PathIcon Data="{Binding Converter={StaticResource TaskListIconValueConverter}}"
 								  Foreground="{Binding Converter={StaticResource TaskListIconForegroundConverter}}" />
```
Comment (Xiaoy312) [inline src/ToDo.UI/Views/HomePage.xaml:108]: if you are re-aligning the xaml (which is fine), make sure the attributes are re-aligned equally as well
& same for line 108+118

**Review by Xiaoy312 — CHANGES_REQUESTED** [review]
please rebase your commits, or at least rename them to something more "elegant"

**Pull request closed without merging**